Utils/src/CandidateTrackFilter.cc

```diff
@@ -0,0 +1,380 @@
+	edm::ESHandle<TransientTrackBuilder> ttBuilder;
+	iSetup.get<TransientTrackRecord>().get("TransientTrackBuilder",ttBuilder);
+
+	//-------------------------------------------------------------------------------------------------
+	// loop over PFCandidates
+	//-------------------------------------------------------------------------------------------------
+	for (size_t i=0; i<pfCandidates->size();i++) {
+		const pat::PackedCandidate pfCand = (*pfCandidates)[i];
+
+		//-------------------------------------------------------------------------------------
+		// pull the track details, if available
+		//-------------------------------------------------------------------------------------
+		if (!pfCand.hasTrackDetails()) continue;
+		auto track = pfCand.pseudoTrack();
+
+		//-------------------------------------------------------------------------------------
+		// place some minimal cuts on the tracks
+		//-------------------------------------------------------------------------------------
+		if (!filterOnTrack(pfCand,track)) continue;
+
+		//-------------------------------------------------------------------------------------
+		// fill the track values and PFCandidate values we'd like to save
+		//-------------------------------------------------------------------------------------
+		auto transientTrack = ttBuilder->build(track);
+		infos.fill(primaryVertex,pfCand,track,transientTrack,true);
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	// loop over lostTracks
+	//-------------------------------------------------------------------------------------------------
+	for (size_t i=0; i<lostTracks->size();i++) {
+		const pat::PackedCandidate lostTrack = (*lostTracks)[i];
+
+		//-------------------------------------------------------------------------------------
+		// pull the track details, if available
+		//-------------------------------------------------------------------------------------
+		if (!lostTrack.hasTrackDetails()) continue;
+		auto track = lostTrack.pseudoTrack();
+
+		//-------------------------------------------------------------------------------------
+		// place some minimal cuts on the tracks
+		//-------------------------------------------------------------------------------------
+		if (!filterOnTrack(lostTrack,track)) continue;
+
+		//-------------------------------------------------------------------------------------
+		// fill the track values and PFCandidate values we'd like to save
+		//-------------------------------------------------------------------------------------
+		auto transientTrack = ttBuilder->build(track);
+		infos.fill(primaryVertex,lostTrack,track,transientTrack,false);
+	}
+
+	//-------------------------------------------------------------------------------------
+	// put track/PFCandidate values back into event
+	//-------------------------------------------------------------------------------------
+	bool result = infos.trks->empty();
```

Comment:
I am concerned about letting this "filter" ever actually reject events. result should be configured as true by default, with a separate option to enable event filtering if it is desired for some reason.

Reply:
Implemented